node_modules/oae-util/lib/modules.js

```diff
@@ -70,31 +71,30 @@ var bootstrapModules = module.exports.bootstrapModules = function(config, callba
  * @api private
  */
 var bootstrapModulesInit = function(modules, config, callback) {
-    var currentModule = 0;
-    var initModule = function() {
-        if (currentModule === modules.length) {
-            callback();
+
+    var MODULE_INIT_FILE = '/lib/init.js';
+    async.mapSeries(modules, function (moduleName, done) {
+        var moduleInitPath = OaeUtil.getNodeModulesDir() + moduleName + MODULE_INIT_FILE;
+
+        if(fs.existsSync(moduleInitPath)) {
+            require(moduleName + MODULE_INIT_FILE)(config, function(err) {
+                if (err) {
+                    log().error(err.stack);
+                    log().error({err: err}, 'Error initializing module %s', moduleName);
+                    return callback(err);
+                }
+                log().info('Initialized module %s', moduleName);
+                done();
+            });
         } else {
-            var module = modules[currentModule];
-            var path = OaeUtil.getNodeModulesDir() + module + '/lib/init.js';
```

Comment:
Should probably double-check this function to make sure it doesn't contain any surprises and then take it out if not needed anymore.